tests/common/fixtures/ptfhost_utils.py

```diff
@@ -515,16 +517,27 @@ def ptf_test_port_map_active_active(ptfhost, tbinfo, duthosts, mux_server_url, d
             target_dut_index = int(list(dut_intf_map.keys())[0])
             target_dut_port = int(list(dut_intf_map.values())[0])
             router_mac = router_macs[target_dut_index]
```

Comment:
When I give "run_tests.sh -n TB_Name -d DUT_2_name -c "qos/test_qos_sai.py" The TB has 3 DUTs, and I have given the second DUT.

This fails:
                # PTF port is mapped to single DUT
                target_dut_index = int(list(dut_intf_map.keys())[0])
                target_dut_port = int(list(dut_intf_map.values())[0])
>               router_mac = router_macs[target_dut_index]
E               IndexError: list index out of range

a_dut_port = 'Ethernet64'
a_dut_port_index = 8
active_active_ports_mux_status = {}
active_dut_map = {}
asic_idx   = 0
disabled_ptf_ports = set([])
dut_intf_map = {'1': 27}
dut_port   = 'Ethernet64'
duthost    = <MultiAsicSonicHost sfd-vt2-lc1>
duthosts   = [<MultiAsicSonicHost sfd-vt2-lc1>]
duts_minigraph_facts = {'sfd-vt2-lc1': [{'deployment_id': None, 'dhcp_servers': [], 'dhcpv6_servers': [], 'forced_mgmt_routes': [], ...}, {'d...t_routes': [], ...}, {'deployment_id': None, 'dhcp_servers': [], 'dhcpv6_servers': [], 'forced_mgmt_routes': [], ...}]}
duts_running_config_facts = {'sfd-vt2-lc1': [{'ACL_TABLE': {'DATAACL': {'policy_desc': u'DATAACL', 'ports': [u'PortChannel101', u'PortChannel103',..._limit_interval': u'600', 'state': u'enabled'}, ...}, 'BGP_DEVICE_GLOBAL': {'STATE': {'tsa_enabled': u'false'}}, ...}]}
idx        = 0
mg_facts   = {'deployment_id': None, 'dhcp_servers': [], 'dhcpv6_servers': [], 'forced_mgmt_routes': [], ...}
mux_server_url = ''
ports_map  = {'0': {'asic_idx': 0, 'dut_port': 'Ethernet0', 'target_dest_mac': 'e8:d3:22:30:22:1a', 'target_dut': [0], ...}, '20': ... '22': {'asic_idx': 1, 'dut_port': 'Ethernet176', 'target_dest_mac': 'e8:d3:22:30:22:1b', 'target_dut': [0], ...}, ...}
ptf_port   = '59'
ptfhost    = <tests.common.devices.ptf.PTFHost object at 0x7f7ae9fee3d0>
router_mac = 'e8:d3:22:30:22:1a'
router_macs = ['e8:d3:22:30:22:18']
target_dut_index = 1
target_dut_port = 27
target_hostname = 'sfd-vt2-lc1'
tbinfo     = {'auto_recover': 'True', 'comment': 'Tests SFD T2 - Vanguard setup', 'conf-name': 'sfd-vt2', 'duts': ['sfd-vt2-lc0', 'sfd-vt2-lc1', 'sfd-vt2-lc2', 'sfd-vt2-sup'], ...}

Reply:
This scenario is not possible to support in 'ptf_test_port_map_active_active' function. I fixed the router_macs based on the duts defined in the testbed (None for duts not present in duthosts),  However, tbinfo['topo'] is defining ports for all the DUTs, and since we are looping through all the ptf ports [Line 502](https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/fixtures/ptfhost_utils.py#L505), it fails to find the corresponding dut_port for ptf ports that are connecting to duts not present in duthosts.